### PR TITLE
Unmaximize icon in titlebar

### PR DESCRIPTION
**Pull request by luximus-hunter**
Continuation of https://github.com/SpikeHD/Dorion/pull/174 :)

Changes the icon of the maximize/unmaximize button to show a windowed mode icon. This way it signals to the user that it will put the window into a floating window again

Floating:

image

Maximized:

image

### CODE REVIEW

File: src-tauri/injection/injection.js [hunk 1]
```diff
@@ -194,4 +198,12 @@ async function ensurePlugins() {
   ]
   
   await Promise.all(promises)
+}
+
+async function setMaximizeIcon() {
+  if (await window.__TAURI__.invoke('is_maximized')) {
```
Comment (SpikeHD) [inline src-tauri/injection/injection.js:204]: You can actually get the state of the window from within the JS api, no need for it's own Tauri command:
https://tauri.app/v1/api/js/window/#ismaximized

`window.__TAURI__.window.appWindow.isMaximized()` should do it

File: src-tauri/injection/injection.js [hunk 2]
```diff
@@ -82,6 +82,10 @@ async function createTopBar() {
 
   appMount.prepend(topbar);
 
+  // Set maximize icon event handler
+  window.addEventListener('resize', setMaximizeIcon)
```
Comment (SpikeHD) [inline src-tauri/injection/injection.js:86]: This is gonna call `setMaximizeIcon` a *bunch* (like, 100 times per 3 seconds of resizing) if the user is just adjusting their window size. While I don't see this realistically being a problem, we should prefer [the Tauri WINDOW_RESIZED event](https://tauri.app/v1/api/js/event#taurievent), which is only called once a resize is completed (and from my testing, only when the state of maximization is changed, so it's perfect!)

Something like this should work:
```js
__TAURI__.event.listen(
  __TAURI__.event.TauriEvent.WINDOW_RESIZED,
  setMaximizeIcon
)
```

**Review by SpikeHD — CHANGES_REQUESTED** [review]

**Comment by SpikeHD** [comment]
Haha, did you by any chance autoformat the entire `injection.js` file? Most recent commit (b3a5e831c48d271b26b9118d8a1395e925b74b4b) adds a bunch of semicolons and double quotes, which don't follow the eslint rules (which the "Check frontend" job should be catching but whatever, another problem for another day).

Running `pnpm codefix` and committing that should bring you back to about what it should be.

**Comment by luximus-hunter** [comment]
Oh shi-

**Comment by SpikeHD** [comment]
Gonna just fix the ESLint styling myself, thank you for the contribution!

**Pull request merged**